packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManagerMapBuffer.java

```diff
@@ -142,6 +142,28 @@ public static boolean isRTL(MapBuffer attributedString) {
         == LayoutDirection.RTL;
   }
 
+  public static Layout.Alignment getTextAlignment(MapBuffer attributedString) {
+    Layout.Alignment alignment = Layout.Alignment.ALIGN_NORMAL;
+
+    MapBuffer fragments = attributedString.getMapBuffer(AS_KEY_FRAGMENTS);
+    if (fragments.getCount() != 0) {
+      MapBuffer fragment = fragments.getMapBuffer(0);
```

Comment:
I think grabbing from first fragment is as good as Paper was doing, but the extra mile here would probably look like translating these to `MetricEffectingSpan`s, then setting the alignment on the `TextPaint` instance. That way, different spans alignment properties will be respected, instead of always using the first.

Reply:
I'm not sure it's that easy unless I misunderstood you or missed something obvious. I've tried `MetricAffectingSpan` with setting the alignment on `TextPaint`, but the alignment is [relative to the origin point](https://developer.android.com/reference/android/graphics/Paint#setTextAlign(android.graphics.Paint.Align)).

<details>
<summary>And this is the result</summary>

<img width="411" alt="Screenshot 2024-04-22 at 16 01 41" src="https://github.com/facebook/react-native/assets/21055725/072bebce-e23f-415a-b578-970579d46b4c">

```jsx
    <Text style={{ backgroundColor: 'cyan', textAlign: 'left' }}>
        <Text style={{ backgroundColor: 'red' }}>Parent Text{'\n'}</Text>
        <Text style={{ backgroundColor: 'red', textAlign: 'center' }}>ABCDEFGHIJKLM{'\n'}</Text>
        <Text style={{ fontWeight: 'bold' }}>Child Text</Text>
        <View style={{width: 50, height: 50, backgroundColor: 'red'}} />
        <Text style={{ fontWeight: 'bold' }}>Child Text</Text>
      </Text>
```

</details>

I've also thought about using `ReplacementSpan` but it doesn't receive the available width in its `draw` nor `getSize` methods.